vyked/registry.py

```diff
@@ -100,15 +100,27 @@ def remove_node(self, node_id):
         thehost = None
         for name, versions in self._registered_services.items():
             for version, instances in versions.items():
+                to_remove = []
                 for instance in instances:
                     host, port, node, service_type = instance
                     if node_id == node:
                         thehost = host
-                        instances.remove(instance)
+                        to_remove.append(instance)
                         try:
                             self.remove_pending_instance(name, version, node_id)
                         except ValueError:
                             pass
+                for instance in to_remove:
+                    instances.remove(instance)
+        for name, versions in self._subscribe_list.items():
+            for version, endpoints in versions.items():
+                for endpoint, subscribers in endpoints.items():
+                    to_remove = []
```

Comment:
would this work?
    to_remove = list(filter(lambda x:node_id == x[4]), subscribers)